cli/ps.go

```diff
@@ -43,39 +56,64 @@ func runPs(args *docopt.Args, client controller.Client) error {
 	if err != nil {
 		return err
 	}
+
+	jobs = prepareAndFilterJobs(jobs, args.Bool["--all"] || args.Bool["-a"], args.String["<type>"])
 	sort.Sort(sortJobs(jobs))
 
+	if args.Bool["--quiet"] || args.Bool["-q"] {
+		printJobsQuiet(jobs)
+	} else {
+		printJobs(jobs, args.Bool["--command"] || args.Bool["-c"])
+	}
+	return nil
+}
+
+func printJobsQuiet(jobs []*ct.Job) {
+	for _, j := range jobs {
+		fmt.Println(j.ID)
+	}
+}
+
+func printJobs(jobs []*ct.Job, commandFlagOn bool) {
 	w := tabWriter()
 	defer w.Flush()
 
 	headers := []interface{}{"ID", "TYPE", "STATE", "CREATED", "RELEASE"}
-	if args.Bool["--command"] {
+	if commandFlagOn {
 		headers = append(headers, "COMMAND")
 	}
 	listRec(w, headers...)
 	for _, j := range jobs {
-		if j.Type == "" {
-			j.Type = "run"
-		}
-		if !args.Bool["--all"] && j.State != ct.JobStateUp && j.State != ct.JobStatePending {
-			continue
-		}
-		id := j.ID
-		if id == "" {
-			id = j.UUID
-		}
```

Comment:
Rather than splitting things out into separate functions, it would be clearer just to do the filtering in the existing loop:

```go
if args.String["--type"] != "" && j.Type != args.String["--type"] {
	continue
}
if args.Bool["--quiet"] {
	fmt.Println(id)
	continue
}
```